.github/workflows/release-porting-guide.yml

```
@@ -0,0 +1,100 @@
+---
+name: Ansible porting guide
+on:
+  workflow_dispatch:
+    inputs:
+      ansible-major-version:
+        description: Major version. For example, 11
+        required: true
+      ansible-build-data-branch:
+        description: >-
+          Release Branch name from the Ansible Build data PR.
+          (e.g. `refs/pull/1/merge`)
+        required: true
+
+jobs:
+  upload-porting-guide:
+    name: Extract the porting guide
+    runs-on: ubuntu-latest
+    env:
+      PORTING_GUIDE_RST: porting_guide_${{ inputs.ansible-major-version }}.rst
+      PORTING_GUIDE_RST_PATH: "./ansible-build-data/${{ inputs.ansible-major-version }}/porting_guide_${{ inputs.ansible-major-version }}.rst"
+      ANSIBLE_RELEASE_BRANCH: ${{ inputs.ansible-build-data-branch }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Check out ansible-build-data
+        uses: actions/checkout@v4
+        with:
+          repository: ansible-community/ansible-build-data
+          ref: ${{ inputs.ansible-build-data-branch }}
+          path: ansible-build-data
+
+      - name: Copy the RST file to the correct path
+        run: cp -v "${PORTING_GUIDE_RST_PATH}" docs/docsite/rst/porting_guides/
```

... and appending it here?
```suggestion
        run: cp -v "${PORTING_GUIDE_RST_PATH}${PORTING_GUIDE_RST}" docs/docsite/rst/porting_guides/
```